client/src/services/userService.js

```diff
@@ -25,13 +25,13 @@ function logout() {
 async function registerUser(userData) {
   try {
     const res = await axios.post(
-      'http://localhost:3000/api/users',
+      '/api/users',
       { userData },
       { header },
     );
-    return res;
+    return JSON.stringify(res.data);
   } catch (error) {
-    console.error(error);
+    return Promise.reject(error);
```

Comment:
Add [react-toastify](https://www.npmjs.com/package/react-toastify) and show error for user. Also 

```suggestion
    return throw error;
```
or with toast

```suggestion
  const { message } = error.response.data

        toast.error(message, {
          autoClose: 2000,
        })
      }
```

Reply:
String 15 unnecessary return